### PR TITLE
release 0.11.0

### LINKED ISSUE #174: 0.11.0 release checklist
### Reasons

Full list of PRs is here https://github.com/owncloud/user_ldap/milestone/11?closed=1 but summarised below.

 - Primarily, we moved sync login into core and out of LDAP https://github.com/owncloud/user_ldap/pull/156  @jvillafanez :x: https://github.com/owncloud/user_ldap/issues/174#issuecomment-382061868
 - Killed all the old ajax.php and moved to controllers https://github.com/owncloud/user_ldap/pull/199
 - Much improved acceptance and UI automated tests
 - UUIDs are not trimmed https://github.com/owncloud/user_ldap/pull/142  @jvillafanez ❗️ https://github.com/owncloud/user_ldap/issues/174#issuecomment-382308470
 - JSON output for list config command https://github.com/owncloud/user_ldap/pull/185  @jvillafanez ✅ (https://github.com/owncloud/user_ldap/issues/174#issuecomment-382314626)
 - Make sure new encryption keys use password with LDAP correctly https://github.com/owncloud/user_ldap/issues/177  @jvillafanez ✅ 

### QA
- [x] Create release branch - @patrickjahns - https://github.com/owncloud/user_ldap/tree/release/0.11.0 
- [x] Create "$version Testplan" @voroyam https://github.com/owncloud/user_ldap/issues/230
- [ ] Test and document in testplan => Final "Go" from QA (set 'QA-ready' label)

### Marketing
- [ ] ~~Check/adapt info.xml stuff~~
- [ ] ~~Check/adapt screenshot~~
- [ ] ~~Blog post or other communication if applicable~~

### Documentation
- [ ] Changes in the docs needed?

### Build & Marketplace release

- [x] Bump version in info.xml
- [x] Update CHANGELOG.md - Change CHANGELOG.md 'Unreleased' to '$version' @patrickjahns
- [x] Create github tag for '$version' @patrickjahns
- [x] Build and sign tarball from github tag '$version' @patrickjahns
- [x] Upload to marketplace @patrickjahns
- [x] Send mail to release-coordination with link to this ticket and to the changelog @patrickjahns
- [x] Rename QA milestone to '$version' @patrickjahns
- [ ] Add a new 'Unreleased' section in CHANGELOG.md @patrickjahns
- [x] Create PR to merge changes back to `master` @patrickjahns

### LINKED ISSUE #230: 0.11.0 Release Test
# 'LDAP' Test Plan

# Testing functionality

# ldap:test-config

- [x] Run occ command ```ldap:test-config``` with a valid configID
- [x] Run occ command ```ldap:test-config``` with a wrong host
- [x] Run occ command ```ldap:test-config``` with empty password
- [x] Run occ command ```ldap:test-config``` with invalid configID

# ldap:show-config

- [x] Run occ command ```ldap:show-config``` with no parameters
- [x] Run occ command ```ldap:show-config``` with valid configID
- [x] Run occ command ```ldap:show-config``` with invalid configID
- [x] Run occ command ```ldap:show-config``` with show-password flag
- [x] Run occ command ```ldap:show-config``` without show-password flag

# ldap:set-config

- [x] Run occ command ```ldap:set-config``` with invalid configID
- [x] Run occ command ```ldap:set-config``` with valid configID

# ldap:search

- [x] Run occ command ```ldap:search``` with valid configuration and no params
- [x] Run occ command ```ldap:search``` with valid configuration and longer limit
- [x] Run occ command ```ldap:search``` with offset multiple of limit (both positive)
- [x] Run occ command ```ldap:search```  name (default configuration)
- [x] Run occ command ```ldap:search``` name (fixed configuration – added “displayName” and/or other attributes in the User Search Attributes field in the wizard)
- [x] Run occ command ```ldap:search``` group (fixed configuration – added “displayName” and/or other attributes in the Group Search Attributes field in the wizard)
- [x] Run occ command ```ldap:search```  Group without search 

# ldap:check-user

- [x] Run occ command ```ldap:check-user``` with a good oC user id 
- [x] Run occ command ```ldap:check-user``` with a wrong oC user id
- [x] Run occ command ```ldap:check-user``` with a good oC user id + 1 disabled configuration
- [x] Run occ command ```ldap:check-user``` Good oc user id + 1 disabled configuration (another one active) + force option
- [x]  Run occ command ```ldap:check-user``` Good oc user id + all disabled configuration + force option; then enable the configuration and recheck

# ldap:create-empty-config

- [x] Run occ command ```ldap:create-empty-config```

# ldap:delete-config

- [x] Run occ command ```ldap:delete-config```

# ldap:show-remnants

- [ ] Run occ command ```ldap:show-remnants``

# ldap:update-group

- [x] Run occ command ```ldap:update-group```

# LDAP quota

**If both LDAP quota field and default LDAP quota are set**

- [x] The user has a value for the LDAP quota attribute
- [x] The user doesn't have a value for the LDAP quota attribute or the value can't be parsed correctly,
- [ ] the user doesn't have a value for the LDAP quota attribute or the value can't be parsed correctly, and the default LDAP quota can't be parsed correctly

**The LDAP quota field is empty and the default LDAP quota is set**

- [x] If default LDAP quota can be parsed correctly
- [ ] If default LDAP quota can't be parsed correctly

**If the LDAP quota field isn't empty and the LDAP default quota isn't set**

- [x] The user has a value for the attribute
- [x] The user hasn't a value for the attribute

**If neither the LDAP quota field nor the default LDAP quota is set**

- [ ] If neither the LDAP quota field nor the default LDAP quota is set

# Several LDAP servers

- [x] Setup and configure 2 LDAP servers correctly
- [ ] Setup and configure 2 LDAP servers correctly, shutdown / disconnect one of them
- [ ] Setup and configure 2 LDAP servers correctly, making sure that at least one user is “shared” in both servers (uid and display name are the same, with different password)
- [x] Setup and configure 2 LDAP servers correctly, making sure that at least one user is “shared” in both servers (uid and display name are the same, with different password), and change in the LDAP wizard the internal username to “uid” (to force a collision)
- [x] Setup and configure 2 LDAP servers correctly, making sure that at least one user is “shared” in both servers (uid and display name are the same, an also with the same password)
- [ ] Setup a Backup (Replica) Host, a Backup (Replica) Port and Disable/Shutdown the Main Server

# Test LDAP properties

- [x] email Set a field as mail in advanced tab
- [x] avatar Login with a user with an avatar field set in LDAP backend
- [x] nested group In advanced tab , check nested groups checkbox
- [x] internalUserAttribute In expert tab , set an attribute as Internal Username

# LDAP Scenarios integrated with external Storage

- [ ] External Storage General Mount an external storage using login credentials save in session
- [ ] External Storage Personal Mount an external storage using login credentials save in session
- [x] WND Mount Point General Mount a WND mountpoint using a user from an Active Directory
- [x] WND Mount Point Personal Mount a WND mountpoint using a user from an Active Directory

# User account table integration

- [x] occ user:sync Run occ user:sync -l
- [x] occ user:sync Run occ user:sync "OCA\User_LDAP\User_Proxy"
- [x] occ user:sync Make changes in oC backend and ReRun occ user:sync "OCA\User_LDAP\User_Proxy" choosing disabling accounts option
- [x] occ user:sync Make changes in oC backend and ReRun occ user:sync "OCA\User_LDAP\User_Proxy" choosing deleting accounts option

# Wizard General

- [x] default configuration: host + port + admin + pass+ base DN + default filters
- [x] configuration: port with string (rest as defaults)
- [x] configuration: wrong credentials
- [x] configuration: from valid config, change to a wrong url and change back to the good ones
- [x] user filter: choose one of the “only those object” select
- [ ] user filter: choose one of the “only those object” select
- [x] user filter: edit raw filter (tested with (|(&(objectclass=inetOrgPerson)(sn=a*))))
- [x] user filter: edit raw filter (tested with (|(&(objectclass=inetOrgPerson)(sn=a*))))
- [x] login filter: checked with LDAP username (uid)
- [x] login filter: checked with LDAP username (uid) and raw filter based on the user filter (tested with (|(&(objectclass=inetOrgPerson)(sn=a*))))
- [x] login filter: checked with LDAP username (uid) and with LDAP email

# Wizard Configuration Server

- [x] 1. Have LDAP running on a standard port
	2. Enter Hostname
	3. Click “Detect Port”
- [x] 1. Have LDAP running on a standard port
	2. Enter Hostname
	3. Enter a wrong port
	4. Click “Detect Port”
- [ ] 1. Have LDAP running on a non-standard port
	2. Enter Hostname
	3. Enter a wrong port 4. Click “Detect Port”
- [ ] 1. Have LDAP running on a non-standard port
	2. Enter Hostname
	3. Enter the correct port
	4. Click “Detect Port”
- [x] 1. Have the host and port configured correctly
	2. Click “Detect Base DN”
- [x] 1. Have the host, port, user dn and password configured correctly
	2. Click “Detect Base DN”
- [x] 1. Have a valid Base DN (and Host, Port, User, Password)
	2. Click “Test Base DN”
- [x] 1. Have Basic tab completed correctly
	2. “Manually enter LDAP filters” is disabled
	3. Move to Users tab
- [x] 1. Have Basic tab completed correctly
	2. “Manually enter LDAP filters” is enabled
	3. Move to Users tab

# Wizard Configuration Users

- [x] 1. “Manually enter LDAP filters” is enabled
	2. The input field for manually writing LDAP filter is shown
- [x] (“raw mode”)
	3. Click on “Edit LDAP Query”
- [x] 1a. “Manually enter LDAP filters” is enabled and assisted mode is active
	1b. or “Manually enter LDAP filters” is disabled
	3. Click on “Edit LDAP Query”
- [x] 1. In Assisted Mode, click the “edit LDAP query” having selected an object class
	2. Click again in the “edit LDAP query”
	3. Select another object class keeping the old class
- [x] 1. Be in Assisted Mode
	2. Open object class multiselect
	3. Change values up to your choice and close it
- [x] 1. Be in Assisted Mode, LDAP server supports memberof,Less than 40 groups available
	2. Open group multiselect
	3. Change values up to your choice and close it
- [x] 1. Be in Assisted Mode, LDAP server supports memberof, At least 40 groups available
- [x] 1. Have the extended group selector
	2. Perform search with the search input field
- [x] 1. Have the extended group selector
	2. Select one or more groups in the “available groups” list
	3. Click the “>” button
- [x] 1. Have the extended group selector
	2. Select one or more groups in the “available groups” list
	3. Click the “<” button
- [x] 1. Have a properly set up filter
	2. Click on “Verify settings and count users”
- [x] 1. Have a filter set up that does not return users
	2. Click on “Verify settings and count users”

# Wizard Configuration login

- [x] 1. Have Basic and Users tab completed correctly
	2. “Manually enter LDAP filters” is disabled
	3. Move to Login Attributes tab
- [x] 1. Have Basic and Users tab completed correctly
	2. “Manually enter LDAP filters” is enabled
	3. Move to Login Attributes tab
- [x] 1. “Manually enter LDAP filters” is enabled
	2. The input field for manually writing LDAP filter is shown (“raw mode”)
	3. Click on “Edit LDAP Query”
- [x] 1a. “Manually enter LDAP filters” is enabled and assisted mode is active
	1b. or “Manually enter LDAP filters” is disabled
	3. Click on “Edit LDAP Query”
- [x] 1. In Assisted Mode, click the “edit LDAP query” having selected an attribute
	2. Click again in the “edit LDAP query”
	3. Select another attribute keeping the old attribute
- [x] 1. Be in Assisted Mode
	2. Open other attributes multiselect
	3. Change values up to your choice and close it
- [x] 1. Be in Assisted Mode
	2. (Un)check LDAP / AD Username
- [x] 1. Be in Assisted Mode
	2. (Un)check LDAP / AD Email address
- [x] 1. Have a properly set up filter
	2. Enter a valid “Test Loginname”
	3. Click on “Verify settings”
- [x] 1. Have a properly set up filter
	2. Enter an invalid “Test Loginname”
	3. Click on “Verify settings”

# Wizard Configuration groups

- [x] 1. Have other tabs completed correctly
	2. “Manually enter LDAP filters” is disabled
	3. Move to Groups tab
- [x] 1. Have other tabs completed correctly
	2. “Manually enter LDAP filters” is enabled
	3. Move to Groups tab
- [x] 1. “Manually enter LDAP filters” is enabled
	2. The input field for manually writing LDAP filter is shown
- [x] (“raw mode”)
	3. Click on “Edit LDAP Query”
- [x] 1a. “Manually enter LDAP filters” is enabled and assisted mode is active
	1b. or “Manually enter LDAP filters” is disabled
	3. Click on “Edit LDAP Query”
- [x] 1. Be in Assisted Mode
	2. Open object class multiselect
	3. Change values up to your choice and close it
- [x] 1. Be in Assisted Mode, Less than 40 groups available
	2. Open group multiselect
	3. Change values up to your choice and close it
- [x] 1. Be in Assisted Mode, at least 40 groups available.
- [x] 1. Have the extended group selector.
	2. Perform search with the search input field
- [x] 1. Have the extended group selector 2. Select one or more groups in the “available groups” list 3. Click the “>” button
- [x] 1. Have the extended group selector
	2. Select one or more groups in the “selected groups” list
	3. Click the “<” button
- [x] 1. Have a properly set up filter
	2. Click on “Verify settings and count groups”
- [x] 1. Have a filter set up that does not return groups
	2. Click on “Verify settings and count groups”

# Wizard Configuration Advanced

- [x] 1. Have other tabs completed correctly
	  2. Go to Advanced Tab
	  3. Be in Connection Settings
- [x] 1. Have other tabs completed correctly
	2. Go to Advanced Tab
	3. Open Directory Settings
- [x] 1. Have other tabs completed correctly
	2. Go to Advanced Tab
	3. Open Special Attributes
- [x] 1. Change anything in advanced tab
	2. Reload
- [x] Click on “Test configuration”
- [x] Click button to clear user mappings
- [x] Click button to clear group mappings

# Wizard Configuration Chooser

- [x] Click on “+” Button
- [x] Click on Clone Button
- [x] Click on Delete Button
- [x] Select a different configuration from the drop down

### PR DESCRIPTION
**Pull request by patrickjahns**
closes #174 
closes #230

### CODE REVIEW

**Comment by phil-davis** [comment]
@patrickjahns master is now up-to-date for acceptance test code.
Rebase this and CI should pass.
